webpack.config.js

```diff
@@ -3,34 +3,40 @@ const HtmlWebpackPlugin = require("html-webpack-plugin");
 
 module.exports = {
   output: {
-    path: path.join(__dirname, "/dist"), // the bundle output path
-    filename: "bundle.js", // the name of the bundle
+    path: path.join(__dirname, "/dist"),
+    filename: 'bundle.[contenthash].js',
   },
   plugins: [
     new HtmlWebpackPlugin({
-      template: "src/index.html", // to import index.html file inside index.js
+      template: "src/index.html",
     }),
   ],
   devServer: {
     port: 3000,
+    historyApiFallback: true
+  },
+  resolve: {
+    extensions: ['.ts', '.js', '.tsx'],
   },
   module: {
     rules: [
       {
-        test: /\.(js|jsx)$/, // .js and .jsx files
-        exclude: /node_modules/, // excluding the node_modules folder
-        use: {
-          loader: "babel-loader",
-        },
+        test: /\.(js|jsx)$/,
+        exclude: /node_modules/,
+        use: "babel-loader",
       },
       {
-        test: /\.(sa|sc|c)ss$/, // styles files
+        test: /\.(sa|sc|c)ss$/,
         use: ["style-loader", "css-loader", "sass-loader"],
       },
       {
-        test: /\.(png|woff|woff2|eot|ttf|svg)$/, // to import images and fonts
-        loader: "url-loader",
-        options: { limit: false },
+        test: /\.(png|woff|woff2|eot|ttf|svg)$/,
+        use: "url-loader"
```

Comment:
DEPREACTED for v5: please consider migrating to [asset modules](https://v4.webpack.js.org/guides/asset-modules/).

Reply:
DEPREACTED for v5: please consider migrating to [asset modules](https://v4.webpack.js.org/guides/asset-modules/).